src/main/scala/LSWidgets.scala

```diff
@@ -126,6 +95,13 @@ class ProcedureWidget(val key: WidgetKey, val state: State, val ws: GUIWorkspace
       override def removeUpdate(e: DocumentEvent): Unit = updateInState(property)
       override def insertUpdate(e: DocumentEvent): Unit = updateInState(property)
     })
+
+  private def boundTextField(i: Int, prop: StringProperty[this.type]) = {
+    val t = new JTextField(i)
```

Comment:
I wasn't sure why you had done it, and I didn't want to introduce a change without a good reason. I'll just remove it.